extension/src/popup/views/AccountCreator.tsx

```diff
@@ -76,70 +94,95 @@ export const AccountCreator = () => {
+                            ? errors.password
+                            : ""
+                        }
+                        {...field}
+                      />
+                    )}
+                  </Field>
                 </FormRow>
                 <FormRow>
-                  <TextField
-                    autoComplete="off"
-                    name="confirmPassword"
-                    placeholder="Confirm password"
-                    type="password"
-                  />
-                  <Error name="confirmPassword" />
+                  <Field name="confirmPassword">
+                    {({ field }: FieldProps) => (
+                      <Input
+                        autoComplete="off"
+                        id="confirm-password-input"
+                        placeholder="Confirm password"
+                        type="password"
+                        error={
+                          authError ||
+                          (errors.confirmPassword && touched.confirmPassword)
+                            ? errors.confirmPassword
+                            : null
+                        }
+                        {...field}
+                      />
+                    )}
+                  </Field>
                 </FormRow>
+
                 <PasswordRequirements />
                 <FormRow>
-                  <CheckboxField
-                    name="termsOfUse"
-                    label={
-                      <span>
-                        I have read and agree to{" "}
-                        <a href="https://stellar.org/terms-of-service">
-                          Terms of Use
-                        </a>{" "}
-                        and{" "}
-                        <a href="https://stellar.org/privacy-policy">
-                          Privacy Policy
-                        </a>
-                      </span>
-                    }
-                  />
-                  <Error name="termsOfUse" />
+                  <Field name="termsOfUse">
+                    {({ field }: FieldProps) => (
+                      <Checkbox
+                        autoComplete="off"
+                        id="termsOfUse-input"
+                        label={
+                          <span>
+                            I have read and agree to{" "}
+                            <TextLink
+                              variant={TextLink.variant.secondary}
+                              href="https://stellar.org/terms-of-service"
+                            >
+                              Terms of Use
+                            </TextLink>
+                          </span>
+                        }
+                        {...field}
+                      />
+                    )}
+                  </Field>
                 </FormRow>
-                <FormRow>
+                {errors.termsOfUse && touched.termsOfUse
```

Comment:
also the checkbox input from SDS doesn't have an `error` prop like the `Input` one does. Should that be added to SDS? 
cc @quietbits @piyalbasu

Reply:
Checkbox component intentionally is built to be flexible. We can't really add an error to every individual checkbox component because they can be in a group, so then we'd want to show an error message for the group.
Maybe we could create a new component `CheckboxWithError` (or a new prop that does this magic internally) or something like that to handle this properly. :thinking: Thoughts?

Reply:
Hmmm interesting - yeah `CheckboxWithError` seems like the way to go. 

There are a couple of other checkboxes that need to be validated in Freighter. I'll keep this in mind as we convert the other ones and come up with a good API for `CheckboxWithError` and add to SDS when I get back from vaca

Reply:
sgtm! I'll add as a TODO and leave for now